src/os_execd/wcom.c

```diff
@@ -304,6 +307,73 @@ size_t wcom_getconfig(const char * section, char ** output) {
     return strlen(*output);
 }
 
+size_t wcom_check_manager_config(char **output) {
+    static const char *daemons[] = {"bin/wazuh-authd", "bin/wazuh-remoted",
+                                    "bin/wazuh-execd", "bin/wazuh-analysisd", "bin/wazuh-logcollector",
+                                    "bin/wazuh-integratord", "bin/wazuh-syscheckd", "bin/wazuh-maild",
+                                    "bin/wazuh-modulesd", "bin/wazuh-clusterd", "bin/wazuh-agentlessd",
+                                    "bin/wazuh-integratord", "bin/wazuh-dbd", "bin/wazuh-csyslogd", NULL
+                                    };
+
+    int ret_val = 0;
+    int result_code = 0;
+    int i;
+    char command_in[PATH_MAX] = {0};
+    char *msg = NULL;
+    char *output_msg = NULL;
+    cJSON *response = cJSON_CreateObject();
+
+    for (i = 0; daemons[i]; i++) {
+        snprintf(command_in, PATH_MAX, "%s %s", daemons[i], "-t");
+        // Exec a command with a timeout of 2000 seconds.
+        if (wm_exec(command_in, &output_msg, &result_code, 2000, NULL) < 0) {
+            if (result_code == EXECVE_ERROR) {
+                mwarn("Path is invalid or file has insufficient permissions. %s", command_in);
+            } else {
+                mwarn("Error executing [%s]", command_in);
+            }
+
+            ret_val = 1;
+            size_t size = snprintf(NULL, 0, "Error executing %s - (%d)", command_in, result_code);
+            os_calloc(size + 1, sizeof(char), msg);
```

Comment:
What happens if the for loop has run once or twice before failing and calling this `os_calloc`? I believe the `wm_strcat`call in line 346 could have allocated memory and stored it in `msg`, making this call a memory leak.

We should try to rethink how we handle this case, memory leak aside, it's a little messy and hard to follow.